aten/src/ATen/native/LinearAlgebra.cpp

```diff
@@ -315,5 +315,50 @@ Tensor& matmul_out(Tensor &result, const Tensor & tensor1, const Tensor & tensor
   return result;
 }
 
+Tensor matrix_power(const Tensor& a, int64_t n) {
+  AT_CHECK(a.dim() >= 2 && at::isFloatingType(a.type().scalarType()),
+           "matrix_power(", a.type(), "{", a.sizes(), "}): expected a tensor "
+           "of floating types with dim at least 2");
+  if (n == 0) {
+    Tensor identities = at::eye(a.size(-2), a.options());
+    std::vector<int64_t> output_size (2, 1);
+    if (a.dim() > 2) {
+      output_size.insert(output_size.begin(), a.sizes().begin(), a.sizes().end() - 2);
+      identities = identities.repeat(output_size);
+    }
+    return identities;
+  } else if (n < 0) {
+    AT_CHECK(a.dim() == 2, "Negative powers for batch matrices are currently not supported")
+    Tensor a_ = at::native::inverse(a);
```

Comment:
I am confused why this is not causing the derivative to be computed. The derivative is defined for the inverse (in `tools/autograd/derivatives.yaml`, and hence this should cause an autodiff based derivative for `n < 0`.

Reply:
The problem is you're calling `at::native::inverse`, when you should call `at::inverse`. There is a long and sordid story about this; see https://github.com/pytorch/pytorch/tree/master/aten/src/ATen/native#can-it-handle-being-passed-variables

Reply:
Thank you @ezyang !